pyuvdata/uvdata/uvdata.py

```diff
@@ -3552,27 +3557,31 @@ def fast_concat(
             this = self
         else:
             this = self.copy()
+        if issubclass(others.__class__, this.__class__):
+            # if this is a UVData object already, stick it in a list
+            others = [others]
```

Comment:
I could imagine a (terrible) situation where this check could fail if say `this` is a HeraData object and `other` is a UVData object. But you can't add those two things together really anyway. But then the iteration where classes are checked again fail but not for the reasons that are desired. 

I wonder if this check should just straight up compare if others is a list/tuple/set (or other well behaved iterable).

Reply:
sure, that makes sense.

Reply:
I agree that we should check to see if `other` is already iterable, and build a single-element list if it's not. The consensus online seems to be to call the `iter()` method inside of a `try` and fail gracefully, so maybe something like this:
```
try:
    others = iter(other)
except TypeError:
    others = iter([other])

for other in others:
    # check that subclasses match, then combine objects
```
I'm wondering if the above might not work if we take multiple passes through the iterator (like first to do a compatibility/object test, then to actually combine objects), but I guess we could always test it and find out. 🙂 

It's also possible to check for `isinstance(other, collections.abs.Iterable)`, but that apparently misses objects that have a `__getitem()__` method but not an `__iter()__` method. Not sure if that's an edge case worth catching or not, but that check might be more robust than building a potentially-single-use iterator.

Reply:
So testing against `collections.abs.Iterable` lets in strings, which I don't think we want. My initial solution to this was to use `isinstance(other, (list, tuple, np.ndarray))`. I guess that could miss some legit things, but I feel like it should be sufficient.

Reply:
I implemented it with the `isinstance(other, (list, tuple, np.ndarray))` for now. Happy to test something else if you want @plaplant, but since UVData object implement an `__iter__` method and somehow match `isinstance(other, collections.abs.Iterable)` neither of those worked out.

Reply:
I think that solution should work fine. yeah having `__iter__` on the object makes it tough.